Library/Homebrew/hardware.rb

```diff
@@ -7,28 +7,35 @@ class CPU
     INTEL_32BIT_ARCHS = [:i386].freeze
     INTEL_64BIT_ARCHS = [:x86_64].freeze
     PPC_32BIT_ARCHS   = [:ppc, :ppc32, :ppc7400, :ppc7450, :ppc970].freeze
-    PPC_64BIT_ARCHS   = [:ppc64].freeze
+    PPC_64BIT_ARCHS   = [:ppc64, :ppc64le, :ppc970].freeze
 
     class << self
       OPTIMIZATION_FLAGS = {
-        native:  "-march=native",
         nehalem: "-march=nehalem",
         core2:   "-march=core2",
         core:    "-march=prescott",
         armv6:   "-march=armv6",
         armv8:   "-march=armv8-a",
+        ppc64:   "-mcpu=powerpc64",
+        ppc64le: "-mcpu=powerpc64le",
       }.freeze
 
       def optimization_flags
-        OPTIMIZATION_FLAGS
+        OPTIMIZATION_FLAGS.tap do |flags|
+          flags[:native] = if ppc?
+            "-mcpu=native"
```

Comment:
Still don't understand the difference for PPC here and why this can't be part of the hash above instead?

Reply:
If I understand your question right, are you asking why this line could not be incorporated to the [line 14](https://github.com/Homebrew/brew/pull/7465/files/63ec2409c39ba12662f744b56c5fa85b665feed9#diff-da65ce29d8f934c7bbb4631f3e1fe672L14)? (that is `native: "-march=native -mcpu=native"`?

Reply:
Ok, I see now. Can this instead be split into a function in `Homebrew/hardware` that takes a parameter e.g. `native` or `Homebrew::EnvConfig.arch` and returns `-mcpu=#{param}` or `-march=#{param}`